libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/service/EvcsService.java

```diff
@@ -0,0 +1,335 @@
+        }
+    }
+
+    @Tracing
+    private void updateEvcsUserVCs(String userId, List<EvcsUpdateUserVCsDto> evcsUserVCsToUpdate)
+            throws EvcsServiceException {
+        try {
+            URI uri =
+                    new URIBuilder(
+                                    configService
+                                            .getSsmParameter(EVCS_APPLICATION_URL)
+                                            .concat("/")
+                                            .concat(userId))
+                            .build();
+            HttpRequest.Builder httpRequestBuilder =
+                    HttpRequest.newBuilder()
+                            .uri(uri)
+                            .method(
+                                    "PATCH",
+                                    HttpRequest.BodyPublishers.ofString(
+                                            OBJECT_MAPPER.writeValueAsString(evcsUserVCsToUpdate)));
+            httpRequestBuilder.header(
+                    X_API_KEY_HEADER, configService.getAppApiKey(EVCS_APP_ID.getPath()));
+            httpRequestBuilder.header(CONTENT_TYPE, APPLICATION_JSON_CHARSET_UTF_8);
+
+            var evcsHttpResponse = sendHttpRequest(httpRequestBuilder.build());
+            checkResponseStatusCode(evcsHttpResponse);
+        } catch (URISyntaxException e) {
+            throw new EvcsServiceException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_CONSTRUCT_EVCS_URI);
+        } catch (JsonProcessingException e) {
+            throw new EvcsServiceException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_PARSE_EVCS_REQUEST_BODY);
+        }
+    }
+
+    @Tracing
+    private void persistEvcsUserVCs(
+            String userId,
+            List<VerifiableCredential> credentials,
+            List<EvcsGetUserVCDto> existingEvcsUserVCs,
+            boolean isF2FIncomplete,
+            boolean is6MFCJourney)
+            throws EvcsServiceException {
+        // AC1 or AC5
+        List<EvcsCreateUserVCsDto> userVCsForEvcs =
+                credentials.stream()
+                        .map(
+                                vc ->
+                                        new EvcsCreateUserVCsDto(
+                                                vc.getVcString(),
+                                                isF2FIncomplete
+                                                        ? EvcsVCState.PENDING_RETURN
+                                                        : EvcsVCState.CURRENT,
+                                                null,
+                                                null))
+                        .toList();
+
+        if (CollectionUtils.isEmpty(existingEvcsUserVCs)
+                || (!CollectionUtils.isEmpty(existingEvcsUserVCs) && isF2FIncomplete)) {
+            // AC1 and AC5
+            createEvcsUserVCs(userId, userVCsForEvcs);
+        } else if (is6MFCJourney) {
```

Comment:
As discussed on the ticket, I'm not sure this should be a separate case here - we should just be comparing the VCs in the current session against the VCs in EVCS.

Reply:
done